### PR TITLE
Add student registration flow

### PR DESCRIPTION
**Pull request by rschamp**
Resolves https://github.com/LLK/scratchr2/issues/3679
Requires PRs in both github.com/LLK/scratchr2 and github.com/LLK/scratch-api

Adds class registration page
Also refactors teacher registration styles so that the registration steps may be reused
I alphabetized routes.json by path since there was not much rhyme or reason to the old order. Redirects are at the bottom.

## Test cases
* Student registration should follow as described in https://github.com/LLK/scratchr2/issues/3679
* Teacher registration should still display correctly

### CODE REVIEW

File: src/views/studentregistration/studentregistration.jsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    },
+    componentDidMount: function () {
+        api({
+            uri: '/classrooms/' + this.props.classroomId + '/' + this.props.classroomToken
+        }, function (err, body, res) {
+            if (err) {
+                return this.setState({
+                    registrationError: this.props.intl.formatMessage({
+                        id: 'studentRegistration.classroomApiGeneralError',
+                        defaultMessage: 'Sorry, we could not find the registration information for this class'
```
Comment (mewtaylor) [inline src/views/studentregistration/studentregistration.jsx]: I don't think we use the `defaultMessage` anymore, just the id.

**Comment by mewtaylor** [comment]
Discussed this offline – outside of the lint fixes needed for travis, as well as a couple small things, this looks great and is ready to go!

**Comment by jwzimmer** [comment]
**Student join flows**
- Used https://staging.scratch.mit.edu/classes/80/register/50c71e56fafa49a783bcf93f4b3fde04
- I've done the CSV & password flows in multiple browsers today for other cases (e.g. https://github.com/LLK/scratch-www/pull/744), not redoing them here.
- The generate link student join workflow is currently broken, blocker, see https://github.com/LLK/scratch-www/issues/770
- Generate link student join workflow has some differences between spec and actual, see https://github.com/LLK/scratch-www/issues/774
- Misaligned validations in IE11 (and Chrome, see teacher registration section below): https://github.com/LLK/scratch-www/issues/775
- If I go to the student join link while logged in as a student with an active reset-password-prompt, I will get redirected from the registration page to the change password page. This is really edge-casey and also doesn't seem likely, so I think it's ok.
- Firefox looks fine
- Safari looks fine
- 

**Teacher registration**
- Created teacher3831
- Lots of misaligned validations due to Chrome 52 vs. 51, blocker: https://github.com/LLK/scratch-www/issues/771
- Spacing issues which @carljbowman & @rschamp already know about... the username text & field are too close together, and the show password checkbox has too much space above it:
screen shot 2016-07-27 at 6 19 24 pm
- Country defaults to united states: https://github.com/LLK/scratch-www/issues/772
- Minor typo "afterscool": https://github.com/LLK/scratch-www/issues/773
- The issues with when validations are triggered (being triggered at different times) still exist but that is a known issue
- Other than the above issues, teacher registration seems to work overall

**Comment by jwzimmer** [comment]
This is not QA-OK since there are outstanding blockers.

**Comment by jwzimmer** [comment]
The blockers were all closed & had fixes/ patches, but I had to re-open #775, so this is not QA-OK.

- https://github.com/LLK/scratch-www/issues/770: The correct card now shows up for registration links
- https://github.com/LLK/scratch-www/issues/775: I found an extant validation misalignment
- https://github.com/LLK/scratch-www/issues/771: In Chrome, validations are aligned correctly
- https://github.com/LLK/scratch-www/issues/696: Followed the last repro steps, did not get an error at the end, created user sdfghdfsfghgfsfdgfh, which I then deleted
- https://github.com/LLK/scratch-www/issues/769:
- https://github.com/LLK/scratch-www/issues/773: Typo is fixed
- https://github.com/LLK/scratch-www/issues/776:

**Pull request merged**